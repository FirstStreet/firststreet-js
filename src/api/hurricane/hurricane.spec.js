const FirstStreet = require('../../__mocks__/FirstStreet');
const hurricaneMock = require('../../__mocks__/hurricaneProperty');

const validMethods = ['getPropertyByFSID', 'getCityByFSID', 'getPropertyByLatLng', 'getCityByLatLng', 'getPropertyByAddress', 'getCityByAddress'];

describe('hurricane', () => {
  const fsf = new FirstStreet('aa.bb.cc', {
    host: 'https://FAKE-HOST-FOR-TESTING.com',
  ***REMOVED***);

  it('should contain valid methods', () => {
    validMethods.forEach((method) => {
      expect(fsf.hurricane[method]).toBeDefined();
    ***REMOVED***);
  ***REMOVED***);

  it('.getPropertyByFSID should get property by ID', async () => {
    const property = await fsf.hurricane.getPropertyByFSID(hurricaneMock.FSID);

    expect(property).toMatchSnapshot();
    expect(property.results[0].floodType).toBe(hurricaneMock.results[0].floodType);
  ***REMOVED***);

  it('.getPropertyByLatLng should get property by lat, lng', async () => {
    const property = await fsf.hurricane.getPropertyByLatLng(-75.6451595133313, -75.6451595133313);

    expect(property).toMatchSnapshot();
    expect(property.results[0].floodType).toBe(hurricaneMock.results[0].floodType);
  ***REMOVED***);

  it('.getPropertyByAddress should get property by address lookup', async () => {
    const property = await fsf.hurricane.getPropertyByAddress('212 appoquin s, middletown, delware');

    expect(property).toMatchSnapshot();
    expect(property.results[0].floodType).toBe(hurricaneMock.results[0].floodType);
  ***REMOVED***);
***REMOVED***);
