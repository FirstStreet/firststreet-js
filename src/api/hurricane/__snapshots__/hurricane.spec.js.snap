// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hurricane .getPropertyByAddress should get property by address lookup 1`] = `
Response {
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.17557,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09307,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.18321,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09554,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.19847,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09909,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.20992,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.10318,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c1",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c5",
        "floodType": "hurricane storm surge",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.90534,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.9098,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.91659,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98473,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.92386,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c3",
        "floodType": "hurricane storm surge",
      },
    ],
  },
}
`;

exports[`hurricane .getPropertyByFSID should get property by ID 1`] = `
Response {
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.17557,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09307,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.18321,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09554,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.19847,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09909,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.20992,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.10318,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c1",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c5",
        "floodType": "hurricane storm surge",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.90534,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.9098,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.91659,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98473,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.92386,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c3",
        "floodType": "hurricane storm surge",
      },
    ],
  },
}
`;

exports[`hurricane .getPropertyByLatLng should get property by lat, lng 1`] = `
Response {
  "data": Object {
    "FSID": 450350219571,
    "results": Array [
      Object {
        "floodData": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.17557,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09307,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.18321,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09554,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.19847,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.09909,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 0,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 0,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.20992,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.10318,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c1",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 11,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 1,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c5",
        "floodType": "hurricane storm surge",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.90534,
              },
            ],
            "year": 2018,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.9098,
              },
            ],
            "year": 2023,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98092,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.91659,
              },
            ],
            "year": 2028,
          },
          Object {
            "data": Array [
              Object {
                "type": "maxdepth",
                "unit": "meters",
                "value": 7,
              },
              Object {
                "type": "lot",
                "unit": "percent",
                "value": 1,
              },
              Object {
                "type": "neighborhood",
                "unit": "percent",
                "value": 0.98473,
              },
              Object {
                "type": "road",
                "unit": "percent",
                "value": 0.92386,
              },
            ],
            "year": 2033,
          },
        ],
        "floodID": "c3",
        "floodType": "hurricane storm surge",
      },
    ],
  },
}
`;
